### ISSUE TITLE
log transformation of size creates unlabeled legend glyphs

### ISSUE THREAD

**Issue by odoublewen** [altair-viz/altair#1905]
I have a plot where I would like a variable encoded in the size of the points, but with a log transform.  I think this is the right way to do it:

```python
import altair as alt
from vega_datasets import data
cars = data.cars()

alt.Chart(cars).mark_point().encode(
    x=alt.X('Horsepower', axis=alt.Axis(labels=False)),
    y=alt.Y('Acceleration', axis=alt.Axis(labels=False)),
    color='Origin',
    size=alt.Size('Displacement', scale=alt.Scale(
        type='log'
    )),
)
```

But the resulting plot has extra, unlabeled legend categories.  This toy example has just two:
![visualization](https://user-images.githubusercontent.com/1421445/72005365-4b210500-3202-11ea-87e7-ae20ea2160a3.png)

...but the real plot I'm making has a lot of them (see screenshot below).  (I provide the toy example above because I feel like it's the same fundamental problem at work?)

The variable in question is an integer (`np.int64`) and has a range of 2 through 7561, with no missing data.  Here is a screenshot of the legend:

<img width="91" alt="Screen Shot 2020-01-08 at 10 38 37 AM" src="https://user-images.githubusercontent.com/1421445/72005707-15305080-3203-11ea-8044-510d0b66eb98.png">

And I see two warnings on the js debug console:

[Warning] WARN – "Scale bindings are currently only supported for scales with unbinned, continuous domains." (vega-lite@4.0.2, line 1, x2)
[Warning] WARN – "Symbol legend count exceeds limit, filtering items." (vega@5.9.0, line 1)

**Comment by jakevdp** [comment]
For some reason base-10 log scales are acting strangely in the current version of vega-lite. It may be a bug. But for data like yours, log scales with different bases might actually work:
```python
import pandas as pd
import numpy as np
import altair as alt

alt.data_transformers.disable_max_rows()

df = pd.DataFrame({
    'x': np.random.randn(7559),
    'y': np.random.randn(7559),
    's': range(2, 7561)
})

alt.Chart(df).mark_point().encode(
    x='x',
    y='y',
    size=alt.Size('s', scale=alt.Scale(type='log', base=2))
)
```
![visualization (93)](https://user-images.githubusercontent.com/781659/72028970-50e60d00-3239-11ea-8601-434556d15595.png)

**Comment by odoublewen** [comment]
THANK YOU!  Yes, base 2 works beautifully.  I see you've created an issue with vega-lite so thank you for that too.

**Comment by odoublewen** [comment]
FYI, it looks like this was fixed with vega >= v5.9.1.   So perhaps this ticket can be closed, although presumably it won't be resolved until Altair is updated to use more recent vega libraries?

**Comment by joelostblom** [comment]
The provided example is working for me on Altair 4.2.0 so I'm closing this, please re-open if you are still seeing the issue.
